docs/proposals/user-namespace.md

```diff
@@ -0,0 +1,206 @@
+
+## Container Runtime Support
+
+### Docker
+
+Docker 1.10 supports [user namespace remapping](https://docs.docker.com/v1.10/engine/reference/commandline/daemon/#daemon-user-namespace-options)
+in the docker daemon.  This allows you to run as the root user in a
+container while running as a non-root user in the host.
+
+In Docker, you may provide the `--userns-remap` option to the daemon
+to enable remapping.  This option will read the `/etc/subuid` and
+the `/etc/subgid` files to determine [how to map UIDs and GIDs
+in the container](https://docs.docker.com/engine/reference/commandline/dockerd/#detailed-information-on-subuidsubgid-ranges).
+
+Additionally, the `HostConfig` for a container now contains a parameter
+that lets you escape the private user namespace and indicate that you
+would like to [use the host's user namespace](https://github.com/docker/engine-api/blob/4290f40c056686fcaa5c9caf02eac1dde9315adf/types/container/host_config.go#L310).
+By using this setting, administrators may run containers outside of the user
+namespace while still requiring other containers to run as a remapped
+user.
+
+### rkt
+
+[rkt supports user namespaces](https://coreos.com/rkt/docs/latest/devel/user-namespaces.html)
+with a systemd-nspawn based implementation.  Each pod started will
+be allocated a random UID range. 
+
+To use private user namespaces in rkt you specify the `--private-users`
+flag on the rkt run command.
+
+
+## Kubernetes Support
+
+As of 1.5 [experimental support](https://github.com/kubernetes/kubernetes/pull/31169)
+has been added which will default the user namespace to host when 
+a pod is using features that require host namespaces (see defaulting scenarios
+described in the design below).  This proposal is 
+building on the experimental work as phase 2 experimental support with 
+the goal of fixing the current known issues.
+
+### Known Issues
+
+These are the known issues (as of writing) that will be hit when
+enabling user namespace remapping in a Kubernetes cluster.
+
+1. Negative oomScoreAdj values fail.  In this case the pause container
+cannot be started and shows an error `write /proc/self/oom_score_adj: permission denied`
+in the daemon log.  Failing in docker 1.10, working in 1.11+
+1. Cannot share the host or a container's network namespace when user
+namespaces are enabled.  [Fixed in 1.11](https://github.com/docker/docker/pull/21383).
+1. Permission denied mounting.  Occurs due to ownership permissions
+in the `/var/lib/kubelet/pods` directory when trying to mount the
+termination logs and secrets.  A workaround (not a solution) is to
+change the remapped GID to the root group.
+
+## Design
+
+### Constraints
+
+* All non-user namespaces have an owning user namespace.  Actions in 
+the non-user namespace require [capabilities that are checked against 
+the user namespace](http://man7.org/linux/man-pages/man7/user_namespaces.7.html).
+This means that when containers share a non-user namespace they must also
```

Comment:
Question answered... you might want to move the constraints stuff up, since it informs some of the text in the prior sections.